### PR TITLE
Pushing down the AGORA_VERSION to DockerHub build

### PR DESCRIPTION
**Pull request by ferencdg**

### CODE REVIEW

**Comment by Geod24** [comment]
Why don't you just run the script and then read the resulting file ?

**Comment by ferencdg** [comment]
> Why don't you just run the script and then read the resulting file ?

I was thinking about that, but then I guess I have to install rdmd before every build...

**Comment by Geod24** [comment]
Good point. Current method is better then.

File: hooks/build [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env sh
+
```
Comment (Geod24) [inline hooks/build:2]: set -xeu
set -o pipefail

File: hooks/build [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+GIT_DESCRIBE_OUT=`git describe`
+
+if [ -n "$GIT_DESCRIBE_OUT"]
+then
+	AGORA_VERSION=$GIT_DESCRIBE_OUT
+fi
+
+echo "Agora version is: $AGORA_VERSION"
+
+docker build --build-arg AGORA_VERSION=$AGORA_VERSION -f $DOCKERFILE_PATH -t $IMAGE_NAME .
```
Comment (Geod24) [inline hooks/build:17]: I guess all those variables are provided by docker hub ?

Comment (ferencdg) [reply hooks/build:17]: yes, I think if we didn't have the build file, then the command that is executed is:
`docker build -f $DOCKERFILE_PATH -t $IMAGE_NAME .`

**Comment by Geod24** [comment]
Looks good. Why is it still a draft ?

**Comment by Geod24** [comment]
I think this broke the dockercloud build, because it does a shallow clone:
```
Executing build hook...
+ set -o pipefail
+ AGORA_VERSION=HEAD
++ git describe
fatal: No names found, cannot describe anything.
+ GIT_DESCRIBE_OUT=
build hook failed! (128)
```

**Comment by ferencdg** [comment]
> I think this broke the dockercloud build, because it does a shallow clone:
> 
> ```
> Executing build hook...
> + set -o pipefail
> + AGORA_VERSION=HEAD
> ++ git describe
> fatal: No names found, cannot describe anything.
> + GIT_DESCRIBE_OUT=
> build hook failed! (128)
> ```

my dockerhub build initially failed, but after I pushed smoe tag back to the remote repository I no longer got that error message

**Comment by Geod24** [comment]
You mean you pushed them to your fork ?

**Comment by ferencdg** [comment]
and it strangely worked after that...

**Comment by Geod24** [comment]
Yeah, docker hub isn't great and tends to pull from the fork...

**Pull request merged**